woocommerce/woocommerce-ios
# Coupons: Enable Save button on Edit Coupon screen only when there are changes made

Closes: #6832  


### Description

This PR adds logic to check for changes on the Edit Coupon screen, so that the Save button is only enabled if there are any changes made to the coupon details.

This also updates the way `populatedCoupon` is created - previously we're using `copy` to update values for the existing coupon. This method ignores nil values, so we cannot remove the expiry date for the coupon by sending nil to the method. The solution is to create a new Coupon manually instead.


### Testing instructions

- Make sure that you have Coupon Management enabled in Settings > Experimental Features.
- Navigate to Menu > Coupons > select a coupon > select the ellipsis button and choose Edit Coupon.
- Notice that the Save button is disabled initially. Editing any of the fields should enable the button.

### Screenshots


https://user-images.githubusercontent.com/5533851/168049177-7f7f1802-f76d-415f-bcf4-295159c88d09.mp4



---
